syntax = "proto3";

package auth;

option go_package = "ctf-service.sso.v1;ssov1";

service Auth {
    rpc Register (RegisterRequest) returns (RegisterResponse);
    rpc Login (LoginRequest) returns (LoginResponse);
    rpc VerifyToken (VerifyTokenRequest) returns (VerifyTokenResponse);
}

message User {
    int64 id = 1;
    string username = 2;
}

message RegisterRequest {
    string username = 1;
    string password = 2;
}

message RegisterResponse {
    int64 id = 1;
}

message LoginRequest {
    string username = 1;
    string password = 2;
}

message LoginResponse {
    string token = 1;
}

message VerifyTokenRequest {
    string token = 1;
}

message VerifyTokenResponse {
    User user = 1;
}